teachertool/src/components/DebugInput.tsx

```diff
@@ -15,9 +15,8 @@ const DebugInput: React.FC<IProps> = ({}) => {
     const [bools, setBools] = useState(true);
 
     const evaluate = async () => {
-        setBools(!bools);
-        await runEvaluateAsync(shareLink, rubric);
-        await loadProjectAsync("hi", bools);
+        await loadProjectMetadataAsync(shareLink);
+        await runEvaluateAsync(rubric);
```

Comment:
I'm thinking that `loadProjectMetadataAsync` should be called in `runEvaluateAsync` rather than here. I don't think the debug container needs context on what the project metadata is, and we can just have `runEvaluateAsync(shareLink, rubric)`, which I think better shows that `runEvaluteAsync` is going to run a rubric on a specific sharelink.

Reply:
I was hesitant to make these calls tightly coupled. I thought there might be scenarios where you want to do one and not the other (like adjust the rubric and re-run evaluate without reloading the share link, or loading a share link in the background while we're still not sure what the rubric will be), and there might also be cases where we just want to simply rely on the share link that's already in state and don't have trivial access to it through some input like we do here (remember that the debug input has it all in one place like this, but that may not always be the case if you're providing share links and then constructing a rubric in separate stages of the UI flow).

It might make sense to combine them down the road if these scenarios never materialize, but until we know a bit more about how that's all going to flow, I personally think it's better to keep them as distinct operations.

Reply:
I see your point @srietkerk. I think this is probably fine as is for the time being. It's an interim implementation. The rubric will move into the app state eventually.

Reply:
For the record, I was not suggesting to get rid of `loadProjectMetadataAsync`, so it would still be available to be called in the scenarios where we might want to load the metadata outside of the evaluate function. 

Otherwise, I understand what you're saying, so I'm okay with keeping it as is, but I do have a bit of a hard time picturing a case where we'd run evaluate and not reload a share link, but maybe I'm thinking of reload too generally.. If we re-evaluate a project, you would still want to know that the project has been loaded and it's in the iframe so you can send the message and get the most accurate response. Whether that be from a backend fetch request or from a fetch in state, you'd still want to make sure that the thing that you're evaluating is the most recent thing, right? I guess I'm saying that I think we _should_ always process the metadata that's being used to make sure that we're evaluating the right project, but maybe that's not necessary.

Reply:
I think you're right. Eventually runEvaluateAsync will pull the rubric from app state.

Reply:
Oh, I wasn't talking about the rubric, I was talking about a project's metadata. I think we will likely pull a rubric from app state, but that's unrelated to my comments.

Reply:
To Sarah's point: I can see the case for calling `loadProjectMetadataAsync` at the start of evaluation, but it does take a while and I'd be hesitant to do it unnecessarily. I'd prefer to leave this as-is for now and make sure we're intentional about each time we call into that transform. If it becomes a pain point later, we can see to it then.

To Eric's point about reading rubric from app state: yeah I could see that happening down the road. Currently the evaluate transform is doing a bit of double-duty, in that it sets the rubric in state and also uses it. Once we have more of a rubric "builder" then I expect these will get pulled out.

For now, I'm inclined to leave this as-is. LMK if that's an issue for anyone.

Reply:
To clarify why updating the metadata take a while, it causes the  iframe to update its url which means we have to wait for the editor to load everything again fresh. We could short-circuit the update if the url is the same, but I think that defeats the purpose you were referring to of fetching any last-minute changes.

Reply:
Sure, that sounds good to me.